app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -252,44 +285,33 @@ fun MapScreen(
           enabled = enableParkingAddition,
           colorLevel = ColorLevel.PRIMARY,
           testTag = "addButton")
+
+      IconButton(
+          icon = Icons.Default.MyLocation,
+          contentDescription = "Recenter on Location",
+          modifier =
+              Modifier.align(Alignment.BottomEnd)
+                  .padding(bottom = 25.dp, end = 16.dp)
+                  .scale(1.2f)
+                  .testTag("recenterButton"),
+          onClick = {
+            mapViewModel.updateTrackingMode(!mapViewModel.isTrackingModeEnable.value)
+
+            mapViewportState.transitionToFollowPuckState(
+                FollowPuckViewportStateOptions.Builder()
+                    .pitch(0.0)
+                    .zoom(maxZoom)
+                    .padding(EdgeInsets(500.0, 100.0, 100.0, 100.0))
+                    .build())
+            mapViewModel.updateTrackingMode(true)
```

Comment:
updateTrackingMode is being called twice in the OnClick is it intended ?

Reply:
I tried removing the first call, and the functionnality is preserved

Reply:
No it is not i will remove it right away !